File: server/src/main/java/de/hsrm/mi/swtpro03/server/network/messages/session/JoinSessionMessage.java
package de.hsrm.mi.swtpro03.server.network.messages.session;

import de.hsrm.mi.swtpro03.server.game.session.SessionRole;

public class JoinSessionMessage {

	public static final String TYPE = "JoinSessionMessage";

	private String sessionID;
	private SessionRole sessionRole;

	public JoinSessionMessage() {
	}

	public JoinSessionMessage(String sessionID, SessionRole sessionRole) {
		this.sessionID = sessionID;
		this.sessionRole = sessionRole;
	}

	public String getSessionID() {
		return sessionID;
	}

	public void setSessionID(String sessionID) {
		this.sessionID = sessionID;
	}

	public SessionRole getSessionRole() {
		return sessionRole;
	}

	public void setSessionRole(SessionRole sessionRole) {
		this.sessionRole = sessionRole;
	}
}

File: client/Client/Network/Messages/Session/JoinSessionMessage.cs
namespace LoadRunnerClient.Messages
{
    public enum SessionRole
    {
        SESSION_OWNER, SESSION_GUEST, SPECTATOR
    }

    public class JoinSessionMessage
    {
        public const string TYPE = "JoinSessionMessage";

        public JoinSessionMessage()
        {
        }

        public JoinSessionMessage(string sessionID, SessionRole sessionRole)
        {
            this._sessionID = sessionID;
            this._sessionRole = sessionRole;
        }

        private string _sessionID;

        public string sessionID
        {
            get => _sessionID;
            set => _sessionID = value;
        }

        private SessionRole _sessionRole;

        public SessionRole sessionRole
        {
            get => _sessionRole;
            set => _sessionRole = value;
        }
    }
}